Future work (#49)
SW705E19/report: sections/discussion/future-work.tex

\\\\
\textit{Calendar system}:
The calendar system would increase the user experience of the application. 
This would make it easier for users to see when certain tutors are available and they would be able to contact the ones who suits their schedule best.
\\\\
\textit{Booking a tutor}:
Booking a tutor is closely related to the calendar system.
The option to book a tutor should be through the tutors calendar. 
The user should be able to request a meeting and then the tutor should have the option to either accept or decline the request.
If the meeting is accepted the tutors and users time slot in the calendar should be registered as occupied, so that other users won't be able to book the tutor in the same time slot.
Users should also not be able to book another tutor in their occupied time slot.
\\\\
\textit{On-site payment}:
On-site payment is one of the less important future work tasks, as tutors and users easily can exchange money with other methods.
It would however still be a nice feature for them to have the option for on-site payment.
\\\\
\textit{Extend the recommender}:

> We would like to improve the recommender system, so that it has a content based and collaborative approach.

Reviewer: ```suggestion
We would like to improve the recommender system, so that it has a content based and collaborative approach, as these could complement each other as described in \autoref{sec:recommender-pros-and-cons}.
```

> Our application can benefit heavily for both approach, as people are expected to have the similar interest. 

Reviewer: ```suggestion
Our application can benefit heavily for both approach.
Currently, the system has issues with cold starts since the collaborative approach struggles with new services, as it needs ratings to create predictions.
Adding a content-based approach in which ratings could be predicted by analyzing similar services could mitigate this.
A complementary approach could also lessen the impact of the strong assumptions associated with collaborative filtering, mainly that a user's taste will not change over time.
```

> However, we do also expect that the quality of the service would change over time, as the tutors gain more experience with their services and hereby would enhance the quality.

Reviewer: Tænker lidt sætning 41 skal væk, det er ikke rigtigt et problem hvis de bare gør deres services bedre, det er mere tænkt som at content-based er bedre til sådan noget som nyheder, hvor de hele tiden skifter og kommer med nye

\section{Future work}
The tasks for future work focuses on completing the tasks in our MoSCoW.
There are 5 tasks that we would like to implement.

\begin{itemize}
    \item Upload and view material
    \item Messaging system
    \item Calendar system
    \item On-site payment
    \item Booking a tutor
    \item Extend the recommender to have content based and collaborative approach.
\end{itemize}
\noindent
\textit{Upload and view material}:
We would like the option for tutors to upload material for users. 
This would make it easier for tutors to share assignments and videos that they have made. 
This was not absolutely essential for the minimum viable product, but would make the user experience for tutors and users better.
\\\\
\textit{Messaging system}: 
It would be nice with an on-site messaging system, so that tutors and users don't have to share personal informations with other users of the site.

> The users and tutors are still able to get in contact with one another, but there is the possibility that their information can be misused.

Reviewer 2: Skriver vi nogle steder hvordan de kan komme i kontakt med hinanden uden et messaging system?

\\\\
\textit{Calendar system}:
The calendar system would increase the user experience of the application. 
This would make it easier for users to see when certain tutors are available and they would be able to contact the ones who suits their schedule best.
\\\\
\textit{Booking a tutor}:
Booking a tutor is closely related to the calendar system.

> The option to book a tutor should be through the tutors calendar. 

Reviewer 2: ```suggestion
The option to book a tutor should be through the tutor's calendar. 
```

The user should be able to request a meeting and then the tutor should have the option to either accept or decline the request.

> If the meeting is accepted the tutors and users time slot in the calendar should be registered as occupied, so that other users won't be able to book the tutor in the same time slot.

Reviewer 2: ```suggestion
If the meeting is accepted the tutor's and user's time slot in the calendar should be registered as occupied, so that other users will not be able to book the tutor in the same time slot.
```

Users should also not be able to book another tutor in their occupied time slot.
\\\\
\textit{On-site payment}:
On-site payment is one of the less important future work tasks, as tutors and users easily can exchange money with other methods.
It would however still be a nice feature for them to have the option for on-site payment.
\\\\
\textit{Extend the recommender}:
We would like to improve the recommender system, so that it has a content based and collaborative approach, as these could complement each other as described in \autoref{sec:recommender-pros-and-cons}.

> Our application can benefit heavily for both approach.

Reviewer 2: ```suggestion
Our application can benefit heavily from both approaches.
```
Har jeg forstået sætningen rigtig?
